janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/cassandra/CqlBridgeRecordReader.java

```diff
@@ -0,0 +1,377 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
Disclaimer: first of all, I should point out I'm not a lawyer and this is not legal advice. 

Having said that, we should **not** be replacing the ASF copyright with a JanusGraph copyright; the copyright stays with the original copyright holder, even (or especially!) in a derivative work.

The hard part here is how to correctly credit the copyright holders and contributors since this single file is being added in isolation to the repository. The best thing would be to have this dependency isolated (e.g., modified in the source library if possible, or brought in as a special third-party library separate from all other code). If neither of those is possible, it would be ideal to identify the history of that file and add those contributors somehow, but it's unclear as to which ones are copyright holders (and hence authors) vs. just contributors.

Re: NOTICE file: while the copyright is ASF on the file, it's not necessarily true that **all** subsequent contributions are also (c) ASF, it just means the initial file was (c) ASF but additional contributions may be (c) those authors. The individual and corporate CLAs that folks sign to contribute to the ASF (same as with JanusGraph) only provide a copyright _license_ not a copyright _assignment_, which means the author retain copyright, but the ASF has license under the Apache 2.0 to redistribute the work.

So, bottom line:

* the copyright header should not be modified from the source file, it should stay as-is
* we should put in a reasonable amount of effort to discover authors/contributors to this file and include them in either AUTHORS / CONTRIBUTORS or NOTICE or both
* in general, it would be best to bring in third-party code into JanusGraph via library dependencies, or well-isolated libraries, rather than file-at-a-time or snippets of code directly into the source tree, as it complicates things, but I understand that it makes it more difficult and inconvenient at times